_rules/active-focus-shortcut-1e9941.md

```diff
@@ -0,0 +1,212 @@
+---
+id: 1e9941
+name: Keyboard shortcut is only active when component has focus
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters for a user interface component, then they are only available when that component has focus.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [HTML document][] with [keyboard shortcuts][keyboard shortcuts].
+
+## Expectation
+
+For each [user interface component][] that is a [descendent][] of the root node of the test target, if [keyboard shortcuts][] are implemented using only [printable characters][], then they are only available when that component has focus.
+
+## Assumptions
+
+This rule assumes as applicable [keyboard shortcuts][keyboard shortcuts] those implemented by the test target [content][]. Any other means (e.g. browser extensions, browser settings, user agents, external browser applications) are not considered.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.1.4: Character Key Shortcuts](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html)
+- [G127 Providing a mechanism to allow users to remap or turn off character key shortcuts](https://www.w3.org/WAI/WCAG21/Techniques/general/G217)
+- [F99 Failure of Success Criterion 2.1.4 due to implementing character key shortcuts that cannot be turned off or remapped](https://www.w3.org/WAI/WCAG21/Techniques/failures/F99)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The [HTML document][] has a [keyboard shortcut][keyboard shortcuts] that uses one [printable][printable characters] and one [non-printable characters][].
+
+```html
+<html>
+  <head>
+    <title>Passed Example 1</title>
+    <script>
+      function shortcut() {
+        document.body.addEventListener('keydown', function(event) {
+          if (event.key === 'i' && event.ctrlKey) {
+            const text = document.getElementById('text');
+            text.className = text.className === 'italic' ? '' : 'italic';
+          }
+        });
+      }
+    </script>
+    <style>
+      .italic {
+        font-style: italic;
+      }
+    </style>
+  </head>
+  <body onload="shortcut();">
+    <div>Press <strong>ctrl+i</strong> to toggle italic format</div>
+    <div id="text">Some text inside the document content</div>

```

Comment:
Actually, I do not think that this document has a UI component. So it passes because there are no UI component, not because the shortcut use non-printable characters.

Reply:
You're right. The instructions are now a button that also toggles italic on and off.